src/validators/file_presence.rs

```diff
@@ -0,0 +1,71 @@
+use crate::validators::issues::{Issue, IssueType, Severity};
+
+const MANDATORY_FILES: &[&str] = &[
+    "agency.txt",
+    "calendar.txt",
+    "calendar_dates.txt",
+    "routes.txt",
+    "stops.txt",
+    "stop_times.txt",
+    "trips.txt",
+];
+
+const OPTIONAL_FILES: &[&str] = &[
+    "fare_attributes.txt",
+    "fare_rules.txt",
+    "feed_info.txt",
+    "frequencies.txt",
+    "shapes.txt",
+];
+
+fn missing_files(raw_gtfs: &gtfs_structures::RawGtfs) -> Vec<Issue> {
+    MANDATORY_FILES
+        .iter()
+        .filter(|m| !raw_gtfs.files.iter().any(|f| f.ends_with(*m)))
```

Comment:
I'm not sure it's important, but in reality you don't want `ends_with` but you want the file name to be equal.
this way `bob_agency.txt` will be considered ok

Reply:
the problem is with subdirectories in the zip library, we only get the full name.

I think it’s good enough like this. Otherwise when would need to be a lot smarter on the reading side.